src/Translator.php

```diff
@@ -0,0 +1,542 @@
+        $site = rtrim($this->txSite, '/');
+        $this->exec("wget $site/dev/tasks/i18nTextCollectorTask?flush=all&merge=1&module=$module");
+    }
+
+    /**
+     * Push source updates to transifex
+     */
+    private function transifexPushSource(): void
+    {
+        $this->log('Pushing updated sources to transifex');
+        if ($this->isDevMode) {
+            $this->log('Not pushing to transifex because TX_DEV_MODE is enabled');
+            return;
+        }
+        foreach ($this->modulePaths as $modulePath) {
+            $this->exec('tx push -s', $modulePath);
+        }
+    }
+
+    /**
+     * Commit changes for all modules
+     */
+    private function gitCommitPushAndPullRequest(): void
+    {
+        $this->log('Committing translations to git');
+        foreach ($this->modulePaths as $modulePath) {
+            // Get endpoint
+            $remote = $this->exec('git config --get remote.origin.url', $modulePath);
+            $remote = str_replace('git@github.com:', 'https://github.com/', $remote);
+            if (!preg_match('#^https://github.com/([^/]+)/(.+?)$#', $remote, $matches)) {
+                throw new Exception("Invalid git remote $remote");
+            }
+            array_shift($matches);
+            list($account, $repo) = $matches;
+            $repo = preg_replace('#\.git$#', '', $repo);
+            $endpoint = "https://api.github.com/repos/$account/$repo/pulls";
+
+            // Add remote
+            if (!in_array('tx-ccs', explode("\n", $this->exec('git remote', $modulePath)))) {
+                $this->exec("git remote add tx-ccs git@github.com:creative-commoners/$repo.git", $modulePath);
+            }
+
+            // Git add all changes
+            $jsPath = $this->getJSLangDirectories($modulePath);
+            $langPath = $this->getLangDirectory($modulePath);
+            foreach (array_merge((array) $jsPath, (array) $langPath) as $path) {
+                if (is_dir($path)) {
+                    $this->exec("git add $path/*", $modulePath);
+                }
+            }
+            $this->exec("git add .tx/config", $modulePath);
+
+            // Check if there's anything to commit
+            $status = $this->exec('git status', $modulePath);
+            if (strpos($status, 'nothing to commit') !== false) {
+                $this->log('Nothing to commit, continuing');
+                continue;
+            }
+
+            // Create new branch
+            $currentBranch = $this->exec('git rev-parse --abbrev-ref HEAD', $modulePath);
+            $time = time();
+            $branch = "pulls/$currentBranch/tx-$time";
+            $this->exec("git checkout -b $branch", $modulePath);
+
+            // Commit changes
+            $title = 'ENH Update translations';
+            $this->exec("git commit -m \"$title\"", $modulePath);
+
+            if ($this->isDevMode) {
+                $this->log('Not pushing changes or creating pull-request because TX_DEV_MODE is enabled');
```

Comment:
```suggestion
                $this->log('Not pushing changes or creating pull-request because TX_DEV_MODE is enabled. A new branch was created.');
```
Just making it really clear what state we're in. Also we're going to get this message (along with others potentially) for every module, so we should probably add a "doing [the thing] for $modulename" at the start of the loop.

Reply:
Updated